tests/test_dynamical_decoupling.py

```diff
@@ -437,6 +437,193 @@ def test_conversion_of_z_pi_2_pulses_at_extremities():
         [2.5e-2, _duration - 2* 2.5e-2, 2.5e-2]))
 
 
+
+def test_conversion_of_pulses_with_arbitrary_rabi_rotations():
```

Comment:
Just a question/comment, no action required (unless you want): do you think we get that much value out of having such a large number of pulses here? The advantage to using a smaller number, like 2 or 3, is that you can write everything out explicitly, which can make it quite a bit simpler to follow the test (and also means you're less likely to get something wrong in the test).

Reply:
Oh, true. I basically only used that large number because I wanted to convince myself that things were working properly. I've now reduced the number of pulses to three in each sequence. In the Rabi/detuning tests I tried to make it so one of the pulses was limited by the upper_limit of the controls, one was limited by the minimum_segment_duration, and one had these two values in perfect balance. Let me know if it's more readable now

Reply:
Nice! Yeah, I find these easier to follow. And it's also more clear that you're specifically triggering all the cases.